substrate/primitives/application-crypto/src/traits.rs

```diff
@@ -133,13 +143,23 @@ pub trait RuntimeAppPublic: Sized {
 	/// Verify that the given signature matches the given message using this public key.
 	fn verify<M: AsRef<[u8]>>(&self, msg: &M, signature: &Self::Signature) -> bool;
 
+	/// Generate proof of possession of the corresponding public key
+	///
+	/// The private key will be requested from the keystore using the given key type.
+	///
+	/// Returns the proof of possession as a signature type or `None` if there is an error.
+	fn generate_pop(&mut self) -> Option<Self::Signature>;
```

Comment:
```suggestion
	fn generate_proof_of_possession(&mut self) -> Option<Self::Signature>;
```

Please let us be explicit.